sherpa/astro/ui/utils.py

```diff
@@ -3154,7 +3227,9 @@ def set_exposure(self, id, exptime=None, bkg_id=None):
         d = self._get_pha_data(id, bkg_id)
         d.exposure = exptime
 
-    def set_backscal(self, id, backscale=None, bkg_id=None):
+    def set_backscal(self, id, backscale=None,
```

Comment:
Is there a reason to just partially type this function?

Reply:
Same below, but again, I'm not marking all of them.

Reply:
Because I do not know the full types that should be used (is it `float`, is it `float | Sequence[float]`, is it `float |  Sequence[float] | np.ndarray`, or ...). 

Typing  of these routines is "easier" once we have the classes better typed (since they define the types).

Here I was focussing on just adding types for "identifiers" (but, as noted above, I got a bit carried away). I htink I'm goong to remove the other types as they are causing too much discussion.